{"name": ["shgo_slsqp_sobol_cube", "shgo_powell_sobol_cube", "shgo_nelder_sobol_cube", "scipy_slsqp_cube", "scipy_powell_cube", "scipy_nelder_cube", "scipy_lbfgsb_cube", "pysot_ei_cube", "pysot_lcb_cube", "pysot_random_cube", "pysot_srbf_cube", "pysot_dycors_cube", "ax_default_cube", "optuna_cmaes_cube", "optuna_tpe_cube", "optuna_random_cube", "optuna_random_cube_clone", "optuna_random_cube_clone_1", "optuna_random_cube_clone_2", "platypus_genetic_cube", "platypus_evolutionary_cube", "platypus_nsgaii_cube", "platypus_cmaes_cube", "platypus_gde3_cube", "platypus_ibea_cube", "platypus_moead_cube", "platypus_omopso_cube", "platypus_smpso_cube", "platypus_spea2_cube", "platypus_epsmoea_cube", "nevergrad_ngopt_cube", "nevergrad_ngopt4_cube", "nevergrad_de_cube", "nevergrad_portfolio_cube", "nevergrad_oneplus_cube", "nevergrad_hammersley_cube", "nevergrad_ngopt8_cube", "hyperopt_tpe_cube", "hyperopt_atpe_cube", "hyperopt_rand_cube", "pymoo_nelder_cube", "pymoo_nsga3_cube", "pymoo_unsga3_cube", "pymoo_pattern_cube", "pymoo_nsga2_cube", "pymoo_unsga3_cube", "skopt_gp_default_cube", "ultraopt_random_cube", "ultraopt_forest_cube", "ultraopt_gbrt_cube", "ultraopt_etpe_cube", "bayesopt_ucb_cube", "bayesopt_ei_cube", "bayesopt_poi_cube", "dlib_cube", "dlib_default_cube", "nlopt_direct_cube", "nlopt_ags_cube", "nlopt_esch_cube", "nlopt_isres_cube", "nlopt_crs2lm_cube", "nlopt_directr_cube", "nlopt_directo_cube", "nlopt_directl_cube", "bobyqa_default_cube", "bobyqa_noise_cube", "hebo_cube_sequential", "hebo_cube_batch", "hebo_sequential_cube", "hebo_batch_cube"], "count": [96, 79, 107, 174, 54, 21, 73, 240, 272, 251, 271, 218, 241, 232, 251, 240, 263, 262, 274, 255, 259, 250, 283, 261, 225, 277, 235, 248, 266, 283, 245, 266, 271, 253, 253, 243, 256, 234, 227, 254, 242, 263, 560, 262, 243, 0, 245, 115, 99, 108, 101, 244, 267, 239, 95, 115, 236, 238, 267, 251, 248, 234, 257, 260, 255, 278, 37, 48, 0, 14], "rating": [1411.0391952886375, 1431.4533300161322, 1369.636903750789, 1757.1969332630406, 1322.9054765514577, 1532.293371489455, 1583.8846142317684, 2142.4169604626136, 2100.088040538597, 1028.298684881085, 2573.8839885964535, 2478.12397275439, 2431.4564264632136, 1999.9311087521573, 2107.213504651891, 1600, 1600, 1600, 1600, 963.8093031302006, 930.8949028367981, 1012.7116127688514, 796.433930939487, 925.4779116281645, 1095.437120390364, 1417.450808943421, 1534.3806086603586, 1229.1471249969438, 712.0123054983521, 1339.3774539511546, 1921.3194287334063, 2013.1132582405148, 1528.2120068968345, 1315.314136666973, 2412.221905202777, 747.2330686918114, 1775.1899431948798, 1614.05297716279, 1656.5109426899612, 1602.8398416250789, 2077.808517337656, 2186.630773651168, 2034.6691136105544, 2242.09569832828, 1025.862960854452, 1600, 2340.5339968891353, 1250.3662688233144, 2069.7229834081727, 1942.8649424301839, 1999.0261280675575, 2398.6837882579075, 2409.189747832145, 2378.4090353798933, 2026.7361359431566, 2100.6788900019696, 2485.225282837785, 2153.693206511578, 1536.3866541597015, 1069.6459374991068, 1401.7132036789053, 2438.4845435135207, 1494.4044556799408, 2660.4654365777633, 2384.259701853173, 2463.945568556532, 1929.3094497418763, 1875.814728828002, 1600, 1818.802252017094], "traceback": ["passing", "passing", "passing", "passing", "Traceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/comparison/eloratings.py\", line 76, in optimizer_game\n    black_best_val, black_best_x, black_feval_count = black(objective, n_trials=n_black_trials,\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 40, in scipy_powell_cube\n    return scipy_cube(objective=objective, n_trials=n_trials, n_dim=n_dim, with_count=with_count, method='powell')\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 29, in scipy_cube\n    result = minimize(_objective, x0=[0]*n_dim, method=options['method'],bounds=bounds, options=options)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/optimize/_minimize.py\", line 614, in minimize\n    return _minimize_powell(fun, x0, args, callback, bounds, **options)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/optimize/optimize.py\", line 3059, in _minimize_powell\n    fx2 = squeeze(func(x2))\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/optimize/optimize.py\", line 464, in function_wrapper\n    return function(np.copy(x), *(wrapper_args + args))\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 27, in _objective\n    return objective(list(x))\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 158, in markowitz_skew_on_cube\n    return realized_something_factory(g=g,u=u)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 121, in realized_something_factory\n    w = cube_to_weights(u)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 108, in cube_to_weights\n    raise ValueError(\"u should be in hypercube\")\nValueError: u should be in hypercube\n", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Black took 228 function evals when instructed to use 228", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Black took 211 function evals when instructed to use 211", "passing", "not yet run", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Black took 211 function evals when instructed to use 211", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Traceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/acq_optimizers/evolution_optimizer.py\", line 119, in optimize\n    res   = minimize(prob, algo, ('n_gen', self.iter), verbose = self.verbose)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/optimize.py\", line 85, in minimize\n    res = algorithm.solve()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 226, in solve\n    self._solve(self.problem)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 321, in _solve\n    self.next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 246, in next\n    self._next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/algorithms/genetic_algorithm.py\", line 93, in _next\n    self.off = self.mating.do(self.problem, self.pop, self.n_offsprings, algorithm=self)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/infill.py\", line 40, in do\n    _off = self.eliminate_duplicates.do(_off, pop, off)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 26, in do\n    pop = pop[~self._do(pop, None, np.full(len(pop), False))]\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 75, in _do\n    D = self.calc_dist(pop, other)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 66, in calc_dist\n    D = cdist(X, X)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/util/misc.py\", line 90, in cdist\n    return scipy.spatial.distance.cdist(A, B, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/spatial/distance.py\", line 2954, in cdist\n    return cdist_fn(XA, XB, out=out, **kwargs)\nValueError: Unsupported dtype object\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/comparison/eloratings.py\", line 76, in optimizer_game\n    black_best_val, black_best_x, black_feval_count = black(objective, n_trials=n_black_trials,\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/hebocube.py\", line 52, in hebo_sequential_cube\n    return hebo_cube_factory(objective=objective, n_trials=n_trials, n_dim=n_dim, with_count=with_count, n_suggestions=1)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/hebocube.py\", line 39, in hebo_cube_factory\n    rec = opt.suggest(n_suggestions=n_suggestions) # <-- don't change this\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/optimizers/hebo.py\", line 124, in suggest\n    rec = opt.optimize(initial_suggest = best_x, fix_input = fix_input).drop_duplicates()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/acq_optimizers/evolution_optimizer.py\", line 125, in optimize\n    res   = minimize(prob, algo, ('n_gen', self.iter), verbose = self.verbose)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/optimize.py\", line 85, in minimize\n    res = algorithm.solve()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 226, in solve\n    self._solve(self.problem)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 321, in _solve\n    self.next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 246, in next\n    self._next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/algorithms/genetic_algorithm.py\", line 93, in _next\n    self.off = self.mating.do(self.problem, self.pop, self.n_offsprings, algorithm=self)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/infill.py\", line 40, in do\n    _off = self.eliminate_duplicates.do(_off, pop, off)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 26, in do\n    pop = pop[~self._do(pop, None, np.full(len(pop), False))]\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 75, in _do\n    D = self.calc_dist(pop, other)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 66, in calc_dist\n    D = cdist(X, X)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/util/misc.py\", line 90, in cdist\n    return scipy.spatial.distance.cdist(A, B, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/spatial/distance.py\", line 2954, in cdist\n    return cdist_fn(XA, XB, out=out, **kwargs)\nValueError: Unsupported dtype object\n", "passing"], "active": [true, true, true, true, true, false, true, true, true, true, true, true, false, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, false, false, true, true]}